packages/react/dialog/src/Dialog.tsx

```diff
@@ -176,7 +181,8 @@ interface DialogOverlayProps extends DialogOverlayImplProps {
 
 const DialogOverlay = React.forwardRef<DialogOverlayElement, DialogOverlayProps>(
   (props: ScopedProps<DialogOverlayProps>, forwardedRef) => {
-    const { forceMount, ...overlayProps } = props;
+    const portalContext = usePortalContext(OVERLAY_NAME, props.__scopeDialog);
+    const { forceMount = portalContext.forceMount, ...overlayProps } = props;
```

Comment:
Why are we keeping the props on the parts? Is it in case they don't use the portal in specific non-modal inline dialogs maybe?

Reply:
> Is it in case they don't use the portal in specific non-modal inline dialogs maybe?
Yeah exactly, cos it's not a required part.